packages/embark/src/lib/modules/blockchain_connector/provider.js

```diff
@@ -129,7 +129,7 @@ class Provider {
               return cb(err);
             }
             if (self.accounts.length) {
-              result.result = self.blockchainAccounts.map(a => a.address);
+              result.result = self.addresses;
```

Comment:
This was intended, but I think it's should be something like so:
```
result.result = self.blockchainAccounts.map(a => a.address || a);
```
Because the account could be just the address, I think